zokrates_core/src/static_analysis/zir_propagation.rs

```diff
@@ -76,6 +76,41 @@ impl<'ast, T: Field> ResultFolder<'ast, T> for ZirPropagator<'ast, T> {
                     }
                 }
             }
+            ZirStatement::IfElse(e, consequence, alternative) => {
+                match self.fold_boolean_expression(e)? {
+                    BooleanExpression::Value(true) => Ok(consequence
```

Comment:
Note for review: If we determine that condition is a constant, we can propagate only the branch that is chosen to save time. This is the reason why this code does not propagate both branches before checking the condition and then picking a branch.